packages/tocco-ui/src/EditableValue/typeEditors/DateAbstract.js

```diff
@@ -76,40 +82,75 @@ class DateAbstract extends React.Component {
     this.wrapper = ref
   }
 
-  toggle() {
-    if (!this.props.readOnly) {
-      this.flatpickr.toggle()
+  hasValue() {
+    return this.props.value && this.props.value.length > 0 && this.props.value.every(v => v)
+  }
+
+  handleOnBlur() {
+    if (this.props.onBlur) {
+      const altValue = this.flatpickr.altInput.value
+      this.props.onBlur(altValue, this.flatpickr.selectedDates, r => this.flatpickr.setDate(r, true))
+    }
+  }
+
+  handleToggleClick() {
+    if (this.props.events && this.props.events) {
+      this.props.events.onFocus()
     }
   }
 
   render() {
-    const spanClass = this.props.readOnly ? 'disabled' : ''
+    const spanClass = classNames('input-group', 'date-edit', {'hidden': this.props.readOnly})
+    const resetClass = classNames('reset', {'hidden': !this.hasValue()})
+
     return (
-      <span
-        className={'input-group date-edit ' + spanClass}
-        onClick={this.toggle.bind(this)}
-        ref={this.refMapper.bind(this)}
-        data-wrap
-      >
-        <input
-          placeholder="Pick date"
-          data-input
-        />
-        <span className="input-group-addon">
-          <span className="glyphicon glyphicon-calendar"/>
+      <span>
+        <span
+          className={spanClass}
+          ref={this.refMapper.bind(this)}
+          data-wrap
+          onBlur={this.handleOnBlur.bind(this)}
+        >
+          <span className="right-addon">
+            <input
+              placeholder={this.props.options.placeholderText}
```

Comment:
`options` is not a required prop. There should either be a null check here or the prop be declared as required.